README.md

```diff
@@ -32,38 +32,31 @@ The Python API is being improved to wrap the algorithms and primitives from the
 
 ## Getting started
 
-### Rapids Memory Manager (RMM)
```

Comment:
RMM is still used and required, no? I think a less prominent mention somewhere might still be good to have. Additionally, I think a quick summary of mdspan/array/etc. might be good to have for newcomers.